src/server/scripts/World/npcs_special.cpp

```diff
@@ -519,6 +519,75 @@ class npc_dancing_flames : public CreatureScript
 };
 
 /*######
+## npc_torch_tossing_target_bunny_controller
+######*/
+
+enum TorchTossingTarget
+{
+    NPC_TORCH_TOSSING_TARGET_BUNNY      = 25535,
+    SPELL_TARGET_INDICATOR              = 45723
+};
+
+class npc_torch_tossing_target_bunny_controller : public CreatureScript
+{
+public:
+    npc_torch_tossing_target_bunny_controller() : CreatureScript("npc_torch_tossing_target_bunny_controller") { }
+
+    struct npc_torch_tossing_target_bunny_controllerAI : public ScriptedAI
+    {
+        npc_torch_tossing_target_bunny_controllerAI(Creature* creature) : ScriptedAI(creature)
+        {
+            _targetTimer = 3000;
+        }
+
+        ObjectGuid DoSearchForTargets()
+        {
+            std::list<Creature*> targets;
+            GuidList targetGuids;
+            me->GetCreatureListWithEntryInGrid(targets, NPC_TORCH_TOSSING_TARGET_BUNNY, 60.0f);
+            for (std::list<Creature*>::const_iterator itr = targets.begin(); itr != targets.end(); ++itr)
+            {
+                if ((*itr)->GetGUID() == _lastTargetGUID)
+                    continue;
+
+                targetGuids.push_back((*itr)->GetGUID());
+            }
+
+            if (!targetGuids.empty())
+            {
+                _lastTargetGUID = Trinity::Containers::SelectRandomContainerElement(targetGuids);
+                targetGuids.clear();
```

Comment:
targetGuids will be deleted as soon as it goes out of scope, so when DoSearchForTargets() exits, so it's not needed to call .clear()